resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -787,6 +791,7 @@ private[spark] class MesosClusterScheduler(
       val toRemove = math.max(retainedDrivers / 10, 1)
       finishedDrivers.trimStart(toRemove)
     }
+    metricsSource.recordFinishedDriver(state)
```

Comment:
I'm not sure about this one. If you go to where "retireDriver" is called, it depends on "TaskState.isFinished", where finished could be any of (FINISHED, FAILED, KILLED, LOST). So, for example, if "supervise" is false and the task failed, it would be recorded here.
I'm assuming this metric is recording drivers in the FINISHED state, not FINISHED or FAILED.

Reply:
Updated the finished case to also log the mesos task state as part of those metrics. So that should result in pretty granular results based on the actual end-state of the 'finished' driver.